contrib/grinder/scripts/utils.py

```diff
@@ -0,0 +1,172 @@
+import time
+import random
+from net.grinder.script.Grinder import grinder
+
+default_config = {
+  'name_fmt': "t4.int.abcdefg.hijklmnop.qrstuvw.xyz.ABCDEFG.HIJKLMNOP.QRSTUVW.XYZ.abcdefg.hijklmnop.qrstuvw.xyz.met.%d",
+  'report_interval': (1000 * 10),
+  'num_tenants': 4,
+  'metrics_per_tenant': 15,
+  'batch_size': 5,
+  'ingest_concurrency': 15,
+  'num_nodes': 1,
+  'url': "http://localhost:19000",
+  'query_url': "http://localhost:20000",
+  'query_concurrency': 10,
+  'max_multiplot_metrics': 10,
+  'search_queries_per_interval': 10,
+  'multiplot_per_interval': 10,
+  'singleplot_per_interval': 10}
+
+units_map = {0: 'minutes',
+             1: 'hours',
+             2: 'days',
+             3: 'months',
+             4: 'years',
+             5: 'decades'}
+
+
+RAND_MAX =  982374239
+
+class ThreadManager(object):
+  #keep track of the various thread types
+  types = []
+
+  @classmethod
+  def add_type(cls, type):
+    cls.types.append(type)
+
+  def convert(self, s):
+    try:
+      return int(s)
+    except:
+      return eval(s)
+
+  def setup_config(self, grinder):
+    #Parse the properties file and update default_config dictionary
+    for entry in grinder.getProperties().entrySet():
+      if entry.value.startswith(".."):
+        continue
+      if entry.key == "grinder.threads":
+        self.tot_threads = self.convert(entry.value)
+      if entry.key.find("concurrency") >= 0:
+        self.concurrent_threads += self.convert(entry.value)
+      if not entry.key.startswith("grinder.bf."):
+        continue
+      k = entry.key.replace("grinder.bf.","")
+      default_config[k] = self.convert(entry.value)
+
+  def __init__(self, grinder):
+    self.tot_threads = 0
+    self.concurrent_threads = 0
+    self.setup_config(grinder)
+    if self.tot_threads != self.concurrent_threads:
```

Comment:
If tot threads and concurrent threads are always equal, why have different config options for them?